frontend/src/app/pages/marker-page/marker-page.component.ts

```diff
@@ -146,18 +146,23 @@ export class MarkerPageComponent implements OnInit {
     }
 
     public sendCompleteLabel(): void {
-        this.sendSelection(new ROISelection3D(<ROISelection2D[]>this.marker.get3dSelection()));
+        this.marker.setFocusable(false);
+        this.dialogService.openInputDialog('Leave a comment (optional)', 'If you\'d like, you can add a comment to the label below:',
+            'Send label').afterClosed().subscribe(comment => {
+            this.sendSelection(new ROISelection3D(<ROISelection2D[]>this.marker.get3dSelection()), comment);
+            this.marker.setFocusable(true);
+        });
     }
 
     public sendEmptyLabel(): void {
-        this.sendSelection(new ROISelection3D());
+        this.sendSelection(new ROISelection3D(), '');
```

Comment:
Hmmm... Your feature may be useful for marking Labels explicitly as empty by adding a comment:

```js
this.sendSelection(new ROISelection3D(), 'This is an empty Label');
```

Of course this will be just a note for humans who will be analysing MedTagger's data ;)